schematic/store/synapse.py

```diff
@@ -51,9 +51,98 @@
 from schematic.exceptions import MissingConfigValueError, AccessCredentialsError
 
 from schematic import CONFIG
+from schematic.utils.general import entity_type_checking
 
 logger = logging.getLogger("Synapse storage")
 
+class ManifestDownload(object):
```

Comment:
@linglp Two things before reviewing more later - hope this helps.

1. My 2nd comment will feed into this, but when you use classes, you always want to start with "why do I need a class"?  I'm not saying this isn't a good use case of classes, but you're currently not utilizing the class structure.
1.  One example for a class is if you have a bunch of functions that take the same arguments. For example, you have  `syn` and `manifest_id` in each of the functions (meaning you are doing an action with `syn` and `manifest_id`.  You could do something like the below. 

    ```
     from dataclasses import dataclass
     @dataclass
     class ManifestDownload:
            syn: synapseclient.Synapse
            manifest_id: str
     ```
     this will allow you to do something like this...
     ```
     md = ManifestDownload(syn=syn, manifest_id=manifest_id)
     md.download_manifest()
     ```

Reply:
Thank you so much Tom. I really like this idea.